fastf1/ergast/interface.py

```diff
@@ -453,6 +462,8 @@ def _build_url(
             selectors.append(f"/fastest/{fastest_rank}")
         if status is not None:
             selectors.append(f"/status/{status}")
+            if endpoint == 'status':
+                endpoint = None
 
         # some special cases: the endpoint may also be used as selector
         # therefore, if the specifier is defined, do not add the endpoint
```

Comment:
Can you please move the four if clauses that you modified down to below this comment, where the other similar cases already are? This way, everything is ordered properly.

Reply:
Most definitely, was under the assumptions the order of the selectors was meaningful. Seems that the final endpoint bears the most weight with everything else being a filter.

Reply:
Ahh, you are right, I didn't think of that. But the correct order is not the one that they have in the code right now. So things are even more incorrect than we thought they are.

Examples:
https://ergast.com/api/f1/constructors/renault/drivers/alonso.json -> driver info
https://ergast.com/api/f1/drivers/alonso/constructors/renault.json -> constructor info

That means that the actual endpoint (+ optional filter that is named equivalently to the enpoint) always needs to come last.
So probably, instead of appending to ``selectors`` and setting ``endpoint=None`` it might be better to not append to ``selectors`` in these cases and set ``endpoint = f"/drivers/{driver}"`` instead. This way, the actual endpoint is always last in the query. Do you agree?

For example:

```

if driver is not None:
    if endpoint == 'driver':
        endpoint = f"/drivers/{driver}"
    else:
        selectors.append(f"/drivers/{driver}")
```